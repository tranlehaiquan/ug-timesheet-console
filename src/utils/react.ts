import * as React from 'react'

type IdentityFunction = <T>(fn: T) => T

export const typedMemo: IdentityFunction = React.memo

export const getDisplayName = <T extends any = any>(WrappedComponent: React.ComponentType<T>): string => {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component'
}
