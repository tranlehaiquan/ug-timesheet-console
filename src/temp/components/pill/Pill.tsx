import * as React from "react";
import { OverflowTooltip } from "../popups/overflow-tooltip/OverflowTooltip";
import { Icon, IconNames } from "../icon/Icon";
import { ITooltipPosition } from "../popups/tooltip/Tooltip";

export interface IPillProps {
  text: string;
  /**
   * Default position is right
   */
  tooltipPosition?: ITooltipPosition;
  iconName?: IconNames;
  onClose?: () => void;
}
/**
 * Creates a Pill. Text will be truncated at 190px and a Tooltip will be enabled
 * If you supply an onClose function, a close icon will appear and call that function on click
 * @requires OverflowTooltip
 * @requires Icon
 */
export const Pill: React.FC<IPillProps> = ({
  iconName,
  tooltipPosition,
  text,
  onClose,
  ...otherProps
}) => {
  return (
    <div
      {...otherProps}
      className="sk-px-3 sk-bg-navy-light sk-font-medium sk-text-xs sk-rounded-full sk-text-white sk-inline-flex sk-leading-loose sk-items-center"
    >
      {iconName && (
        <Icon name={iconName} size={12} className="sk-text-white sk-mr-2" />
      )}
      <OverflowTooltip maxWidth={190} position={tooltipPosition}>
        {text}
      </OverflowTooltip>
      {onClose && (
        <Icon
          data-sk-name="pill-close"
          size={8}
          name="remove"
          className="sk-text-grey-light hover:sk-text-white sk-cursor-pointer sk-ml-2"
          onClick={onClose}
        />
      )}
    </div>
  );
};
