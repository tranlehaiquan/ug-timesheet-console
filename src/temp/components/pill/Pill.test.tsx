import * as React from 'react'
import { shallow, render } from 'enzyme'

import { Pill } from './Pill'

describe('Pill', () => {
  test('render', () => {
    const wrapper = shallow(<Pill iconName="urgent" text="Urgent" onClose={ jest.fn() } />)
    expect(wrapper).toMatchSnapshot()
  })

  test('shows specified icon', () => {
    const wrapper = shallow(<Pill iconName="urgent" text="Urgent" />)
    const Icon = wrapper.find('Icon')
    expect(Icon.length).toEqual(1)
    expect(Icon.prop('name')).toEqual('urgent')
  })

  test('adds close icon when required', () => {
    const closeSpy = jest.fn()
    const wrapper = shallow(<Pill onClose={ closeSpy } text="Option" />)
    const CloseIcon = wrapper.find('[data-sk-name="pill-close"]')
    expect(CloseIcon.length).toEqual(1)
    CloseIcon.simulate('click')
    expect(closeSpy).toBeCalled()
  })

  test('restricts text after 190px', () => {
    const wrapper = shallow(<Pill text="Blah blah blaaaaaaaaaaaaaahhhhhhhhhhhhhhhhh" />)
    expect(wrapper.find('OverflowTooltip').prop('maxWidth')).toEqual(190)
  })

  test('sets tooltip position', () => {
    const wrapper = shallow(<Pill text="Blah blah blaaaaaaaaaaaaaahhhhhhhhhhhhhhhhh" tooltipPosition="top" />)
    expect(wrapper.find('OverflowTooltip').prop('position')).toEqual('top')
  })
})
