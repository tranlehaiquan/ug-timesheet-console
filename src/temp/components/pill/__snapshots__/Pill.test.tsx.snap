// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill render 1`] = `
<div
  className="sk-px-3 sk-bg-navy-light sk-font-medium sk-text-xs sk-rounded-full sk-text-white sk-inline-flex sk-leading-loose sk-items-center"
>
  <Icon
    className="sk-text-white sk-mr-2"
    name="urgent"
    size={12}
  />
  <OverflowTooltip
    maxWidth={190}
    position="right"
  >
    Urgent
  </OverflowTooltip>
  <Icon
    className="sk-text-grey-light hover:sk-text-white sk-cursor-pointer sk-ml-2"
    data-sk-name="pill-close"
    name="remove"
    onClick={[MockFunction]}
    size={8}
  />
</div>
`;
